app/scripts/filters/canI.js

```diff
@@ -2,15 +2,22 @@
 
 angular
   .module('openshiftConsole')
-  .filter('canIDoAny', function(canIFilter) {
-    var resourceRulesMap = {
+  .filter('canIDoAny', function(APIService, canIFilter) {
+    // Top level keys are representing pages.
+    // Within each page may one or more resources.
+    // canIDoAny returns truthy if the user passes the canIFilter
+    // for at least one of the resources listed in the page.
+    // TODO: this is confusing, we should come back & fix
+    // the view should likely drive what it is requesting by passing an
+    // array to the filter
+    var pageRulesMap = {
       'buildConfigs': [
         {group: '', resource: 'buildconfigs',             verbs: ['delete', 'update']},
         {group: '', resource: 'buildconfigs/instantiate', verbs: ['create']}
       ],
       'builds': [
-        {group: '', resource: 'builds/clone', verbs: ['create']},
-        {group: '', resource: 'builds',       verbs: ['delete', 'update']}
+        _.extend({}, APIService.getPreferredVersion('builds/clone'), {verbs: ['create']}),
+        _.extend({}, APIService.getPreferredVersion('builds'), {verbs: ['delete', 'update']})
```

Comment:
We should use `_.assign` instead of `_.extend`. I think we only want own properties.